### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by jonschlinkert**
@doowb please review

### CODE REVIEW

**Comment by doowb** [comment]
Looks good to me!

I think the only thing I would suggest, is a short explanation that returning `false` from the `.filter` function is what excludes the file. Even though I assumed by the name that it works like an Array filter, I wasn't positive if it should return `true` to say, "yes, exclude this file", or `false` to say, "this file should be filtered out".

**Comment by jonschlinkert** [comment]
> is a short explanation that returning `false` from the `.filter` function is what excludes the file.

Are you saying I should change the function to do that?

File: index.js [hunk 1]
```diff
@@ -1,73 +1,115 @@
...
+const requires = (dir, obj = {}, options = {}) => {
+  if (typeof options === 'function') {
+    options = { filter: options };
   }
 
-  var dirs = fs.readdirSync(dir);
-  var len = dirs.length;
-  var idx = -1;
-  var res = {}
+  const opts = { case: 'auto', filter: file => file.name !== 'index.js', ...options };
```
Comment (jonschlinkert) [inline index.js:15]: @doowb here it's saying "return true for any file that is not `index.js`"

File: index.js [hunk 2]
```diff
@@ -1,73 +1,115 @@
...
+    if (file.isFile() && !exts[file.extname]) continue;
+
+    const names = namify(file, opts);
+    const fns = typeof opts.case !== 'function'
+      ? [].concat(opts.case).map(k => names[k])
+      : names;
+
+    const keys = fns.map(fn => typeof fn === 'function' ? fn() : fn);
+
+    if (opts.filter(file, keys, exportsObj) === false) continue;
```
Comment (jonschlinkert) [inline index.js:63]: @doowb Here it's doing the same, it's just not being called on an array.

**Comment by jonschlinkert** [comment]
> returning `false` from the `.filter` function is what excludes the file. 

This is how filter always works. Return true for files to keep, and false for files to exclude. 

K I think based on your comment I'll go ahead and merge. Thanks!

File: .verb.md [hunk 3]
```diff
@@ -1,15 +1,116 @@
...
+
+### .filter
+
+Function to exclude files from the result.
+
+**Type:** `function`
+
+**Default:** Excludes files named `index.js`
+
+**Example**
```
Comment (doowb) [inline .verb.md:69]: @jonschlinkert I just meant adding something here like:

> The following example shows how to exclude files named `a.js`.

**Pull request merged**